src/GenerateB.c

```diff
@@ -236,6 +307,21 @@ void GenerateB (const enum incpol which,   // x - or y polarized incident light
 	const double *ex; // coordinate axis of the beam reference frame
 	double ey[3];
 	double r1[3];
+	/* complex wave amplitudes of secondary waves (with phase relative to particle center);
+	 * The transmitted wave can be inhomogeneous wave (when msub is complex), then eIncTran (e) is normalized
+	 * counter-intuitively. Before multiplying by tc, it satisfies (e,e)=1!=||e||^2. This normalization is consistent
+	 * with used formulae for transmission coefficients. So this transmission coefficient is not (generally) equal to
+	 * the ratio of amplitudes of the electric fields. In particular, when E=E0*e, ||E||!=|E0|*||e||, where
+	 * ||e||^2=(e,e*)=|e_x|^2+|e_y|^2+|e_z|^2=1
+	 */
+	doublecomplex eIncRefl[3],eIncTran[3];
+#ifndef NO_FORTRAN
+	double ro;
+	int n1,q; // for Bessel beams
+	doublecomplex vort;  // vortex phase of Bessel beam rotated on 90 deg
+	doublecomplex fn[5]; // for general functions f(n,ro,phi) of Bessel beams (fn-2, fn-1, fn, fn+1, fn+2 respectively)
+	double phi,arg,td1[abs(n0)+3],td2[abs(n0)+3],jn1[abs(n0)+3]; // for Bessel beams
```

Comment:
What will happen if n=10^6 or 10^9. Should be specify some limit in input parameters? Or use alternative code for large arguments (to save memory)? Or at least not recommend the user to use orders larger than some number.